tests/ui/job-view/Filtering_test.jsx

```diff
@@ -0,0 +1,275 @@
+import React from 'react';
+import { fetchMock } from 'fetch-mock';
+import {
+  render,
+  cleanup,
+  fireEvent,
+  waitForElement,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+
+import App from '../../../ui/job-view/App';
+import reposFixture from '../mock/repositories';
+import pushListFixture from '../mock/push_list';
+import { getApiUrl } from '../../../ui/helpers/url';
+import {
+  getProjectUrl,
+  replaceLocation,
+  setUrlParam,
+} from '../../../ui/helpers/location';
+import jobListFixtureOne from '../mock/job_list/job_1';
+import jobMap from '../mock/job_map';
+
+describe('Filtering', () => {
+  const repoName = 'autoland';
+
+  beforeAll(() => {
+    fetchMock.get('/revision.txt', []);
+    fetchMock.get(getApiUrl('/repository/'), reposFixture);
+    fetchMock.get(getApiUrl('/user/'), []);
+    fetchMock.get(getApiUrl('/failureclassification/'), []);
+    fetchMock.get('begin:https://treestatus.mozilla-releng.net/trees/', {
```

Comment:
The ``begin:`` is basically the same as ``startsWith`` in JS.  If it gets any ``fetch`` call that begins with that string, it'll return this value.  Since I'm not testing treestatus, I'm just returning dummy data, regardless of which tree it tries to get status for.